src/main/scala/vsys/api/http/contract/ContractApiRoute.scala

```diff
@@ -35,46 +43,121 @@ case class ContractApiRoute (settings: RestAPISettings, wallet: Wallet, utx: Utx
-      , ContractNotExists(contractName))
-    complete(result)
+  def content: Route = (get & path("content" / Segment)) { encoded =>
+    ByteStr.decodeBase58(encoded) match {
+      case Success(id) => state.contractContent(id) match {
+        case Some((h, txId, ct)) => complete(Json.obj("transactionId" -> txId.base58) ++ ct.json ++ Json.obj("height" -> JsNumber(h)))
+        case None => complete(ContractNotExists)
+      }
+      case _ => complete(InvalidAddress)
+    }
   }
 
-  @Path("/enable")
-  @ApiOperation(value = "Enables a contract",
-    httpMethod = "POST",
-    produces = "application/json",
-    consumes = "application/json")
+  @Path("/info/{contractId}")
+  @ApiOperation(value = "Info", notes = "Get contract info associated with a contract id.", httpMethod = "GET")
   @ApiImplicitParams(Array(
-    new ApiImplicitParam(
-      name = "body",
-      value = "Json with data",
-      required = true,
-      paramType = "body",
-      dataType = "vsys.api.http.contract.ChangeContractStatusRequest",
-      defaultValue = "{\n\t\"contractName\": \"contractname\",\n\t\"sender\": \"3Mx2afTZ2KbRrLNbytyzTtXukZvqEB8SkW7\",\n\t\"fee\": 100000,\n\t\"feeScale\": 100\n}"
-    )
+    new ApiImplicitParam(name = "contractId", value = "Contract ID", required = true, dataType = "string", paramType = "path")
   ))
-  @ApiResponses(Array(new ApiResponse(code = 200, message = "Json with response or error")))
-  def enable: Route = processRequest("enable", (t: ChangeContractStatusRequest) => doBroadcast(TransactionFactory.changeContractStatus(t, ChangeContractStatusAction.Enable, wallet, time)))
+  def info: Route = (get & path("info" / Segment)) { contractId =>
+    complete(infoJson(contractId))
+  }
+
+  private def infoJson(contractIdStr: String): Either[ApiError, JsObject] = {
+    ByteStr.decodeBase58(contractIdStr) match {
+      case Success(id) => state.contractContent(id) match {
+        case Some((h, txId, ct)) => Right(Json.obj(
+          "contractId" -> contractIdStr,
+          "transactionId" -> txId.base58,
+          "info" -> JsArray((ct.stateVar, paraFromBytes(ct.textual.last)).zipped.map { (a, b) =>
+            (state.contractInfo(ByteStr(id.arr ++ Array(a(0)))), b) }.filter(_._1.isDefined).map { a => a._1.get.json ++ Json.obj("name" -> a._2) }),
+          "height" -> JsNumber(h))
+        )
+        case None => Left(ContractNotExists)
+      }
+      case _ => Left(InvalidAddress)
+    }
+  }
+
+  @Path("/tokenInfo/{tokenId}")
+  @ApiOperation(value = "Token's Info", notes = "Token's info by given token", httpMethod = "Get")
+  @ApiImplicitParams(Array(
+    new ApiImplicitParam(name = "tokenId", value = "Token ID", required = true, dataType = "string", paramType = "path")
+  ))
+  def tokenInfo: Route = (get & path("tokenInfo" / Segment)) { tokenId =>
+    ByteStr.decodeBase58(tokenId) match {
+      case Success(id) => {
+        val maxKey = ByteStr(id.arr ++ Array(0.toByte))
```

Comment:
can keys be deleted?  what are these keys?

Reply:
these keys are default key-format to record the token information.
key = tokenId + one byte
include max supply, current total supply, unity, description of token